addons/maintainence-agent/maintenance_mode_controller.go

```diff
@@ -122,62 +130,88 @@ func (r *MaintenanceModeReconciler) startMaintenanceActions(ctx context.Context,
-	for _, cephCluster := range clusters {
-		deployments, err := GetDeploymentsStartingWith(ctx, r.SpokeClient, cephCluster.Namespace, RBDMirrorDeploymentNamePrefix)
-		if err != nil {
-			return ctrl.Result{}, err
-		}
+func (r *MaintenanceModeReconciler) scaleRBDMirrorDeployment(ctx context.Context, cluster rookv1.CephCluster, replicas int) (ctrl.Result, error) {
+	isMirroringEnabled, err := r.isStorageClusterMirroringEnabled(ctx, cluster)
+	if err != nil {
+		klog.Errorf("error occurred on checking mirroring enablement on storagecluster for cephcluster %s", cluster.Name)
+		return ctrl.Result{}, err
+	}
+	if !isMirroringEnabled {
+		klog.Info("storagecluster mirroring is not enabled yet. please enable it manually or wait for it to be enabled to perform further maintenance actions")
+		return ctrl.Result{Requeue: true}, nil
+	}
+	deployments, err := GetDeploymentsStartingWith(ctx, r.SpokeClient, cluster.Namespace, RBDMirrorDeploymentNamePrefix)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
 
-		for _, deploymentName := range deployments {
-			err = scaleDeployment(ctx, r.SpokeClient, deploymentName, cephCluster.Namespace, int32(replicas))
-			return ctrl.Result{}, err
-		}
+	for _, deploymentName := range deployments {
+		err = utils.ScaleDeployment(ctx, r.SpokeClient, deploymentName, cluster.Namespace, int32(replicas))
+		return ctrl.Result{}, err
 	}
 	return ctrl.Result{}, nil
 }
 
-func (r *MaintenanceModeReconciler) updateStatus(ctx context.Context, mmode ramenv1alpha1.MaintenanceMode, mode ramenv1alpha1.MMode, state ramenv1alpha1.MModeState, err error) (ctrl.Result, error) {
+// Checks the mirroring status of the storagecluster. Deployment will not be present if mirroring is not enabled.
+func (r *MaintenanceModeReconciler) isStorageClusterMirroringEnabled(ctx context.Context, cephCluster rookv1.CephCluster) (bool, error) {
+	var sc ocsv1.StorageCluster
+	scName := getOwnerName(cephCluster)
+	if scName == "" {
+		return false, fmt.Errorf("no storagecluster found for given cephcluster %s", cephCluster.Name)
+	}
+	err := r.SpokeClient.Get(ctx, types.NamespacedName{Name: scName, Namespace: cephCluster.Namespace}, &sc)
+	if err != nil {
+		klog.Errorf("no storagecluster found for given cephcluster %s", cephCluster.Name)
+		return false, err
+	}
+	return true, nil
+}
+
+func getOwnerName(cluster rookv1.CephCluster) string {
+	for _, owner := range cluster.OwnerReferences {
+		if owner.Kind != "StorageCluster" {
+			continue
+		}
+		return owner.Name
+	}
+	return ""
+}
+
+func SetStatus(ctx context.Context, mmode ramenv1alpha1.MaintenanceMode, mode ramenv1alpha1.MMode, state ramenv1alpha1.MModeState, err error) {
 	klog.Infof("mode: %s, status: %s, generation: %d, error: %v ", mode, state, mmode.Generation, err)
 	mmode.Status.State = state
 	mmode.Status.ObservedGeneration = mmode.Generation
-	var conditions []metav1.Condition
+	var conditions = make([]metav1.Condition, 0)
```

Comment:
(nit) we should just be able to reuse mmode.Status.Conditions here.

Reply:
Not sure why but i am using this as i was getting the following error 
> MaintenanceMode.ramendr.openshift.io "mmode-2" is invalid: status.conditions: Invalid value: "array": status.conditions in body must be of type object: "array"

If you have any idea why this happens, please let me know

Reply:
No worries, just a not at present.